csharp/private/rules/create_net_workspace.bzl

```diff
@@ -0,0 +1,31 @@
+"""A rule that sets up the @net workspace.
+
+local_repository doesn't work well because it takes a path that is either
+absolute (yuck) or relative to our users WORKSPACE file. Instead we use a
+custom workspace rule.
+"""
+
+def _create_net_workspace_impl(ctx):
+  # The "trick" here is that repository rules each create a workspace (named by
+  # their name attr). So when we create files with ctx it's inside that
+  # workspace.
```

Comment:
I started by doing the dumb thing and using a sacrificial NuGet package, but then felt bad and went digging in [around `repository_ctx.extract`](https://docs.bazel.build/versions/master/skylark/lib/repository_ctx.html#extract) That and looking at how [`http_archive` works](https://github.com/bazelbuild/bazel/blob/master/tools/build_defs/repo/http.bzl#L59) led me to [the `workspace_and_buildfile` function](https://github.com/bazelbuild/bazel/blob/master/tools/build_defs/repo/utils.bzl#L31) which was the basis for how this works.